### ISSUE TITLE
Is Export and Load table operation suppose to work synchronously ? 

### ISSUE THREAD

**Issue by Dsiddhesh** [googleapis/nodejs-bigquery#850]
Hi Team,

Could you please let me know  if the Load and Export Table operations are suppose to be synchronous or asynchronous ? I don't see anything mentioned in the official [documentation](https://cloud.google.com/bigquery/docs/exporting-data).  I tried export and it appears like it is synchronous operation it waits for internal thread to complete. What can be done to make it asynchronous ? use explicit Job and execute the desired operation ?

**Comment by stephenplusplus** [comment]
When you call "table.extract()", we make an API call. That call returns the results of the request within it, as opposed to returning an Operation that requires further pinging for status updates. Still, our API is asynchronous-- either the callback or Promise should be resolved after the API gives us that response.

Could you elaborate on what you are experiencing, with code included? We have an issue template that can help us get on the same page more quickly: https://raw.githubusercontent.com/googleapis/nodejs-bigquery/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by steffnay** [comment]
@Dsiddhesh I am closing this issue, but please re-open with more information about your issue if you would like further investigation.

**Comment by Dsiddhesh** [comment]
Hi @steffnay , sorry for delayed response.  I don't see option to reopen the issue. I tried POC to export data from Table to GCS using promise. I get response after  the operation completes.  I have attached POC that I am trying using promise. I see belwo output. See delay of 1 minute.  I am exporting Table of size 1.5 GB. 

D:\BigQuery\POC>node exporttoGCS.js
**2020-12-01T12:32:36.780Z
2020-12-01T12:33:55.089Z
[exporttoGCS (2).zip](https://github.com/googleapis/nodejs-bigquery/files/5622626/exporttoGCS.2.zip)

**
*****resp {"kind":"bigquery#job","etag":"SXdVnDMt9Ji0DXCd3iQw==","id":"project1-279307:US.892fccf2-2680-4d66-b6eb-372c8776d5dd","selfLink":"https://bigquery.googleapis.com/bigquery/v2/projects/bigqueryproject1-279307/jobs/89fccf2-2680-4d66-b6eb-378776d5dd?location=US","user_email":"dev1-761@project1-279307.iam.gserviceaccount.com","configuration":{"extract":{"sourceTable":{"projectId":"project1-279307","datasetId":"testdataset","tableId":"tabledata_1gb"},"destinationUri":"gs://pslbucket/TestExport*.CSV","destinationUris":["gs://pslbucket/TestExport*.CSV"],"destinationFormat":"CSV"},"jobType":"EXTRACT"},"jobReference":{"projectId":"project1-279307","jobId":"892fccf2-2680-4d66-b6eb-372c8776d5dd","location":"US"},"statistics":{"creationTime":"1606825959239","startTime":"1606825959639","endTime":"1606826034472","completionRatio":1,"extract":{"destinationUriFileCounts":["2"],"inputBytes":"2480055463"},"totalSlotMs":"296787","reservationUsage":[{"name":"default-pipeline","slotMs":"296787"}],"reservation_id":"default-pipeline"},"status":{"state":"DONE"}}

**Comment by Dsiddhesh** [comment]
Hi @steffnay , Shall i open new issue/question ?

**Comment by stephenplusplus** [comment]
Sorry about that. I reopened it for us.

I'm not entirely clear on the issue you're experiencing, so I'll state some things that hopefully hit close enough that we can figure it out together.

We have two methods, `extract` and `createExtractJob`:

- `createExtractJob` instructs the server to create and immediately begin an extract job. It returns a `Job` object to the user which can poll the server to receive updates on the extract job.
- `extract` is a convenience method that calls `createExtractJob` internally, waits on the completed status of the server's extract operation, and returns it to the user.

In JavaScript terminology, both of these are asynchronous methods. They are both non-blocking operations to the sequence of the code in your JavaScript file. The example code you provided does not appear to demonstrate thread blocking behavior, just that the operation took that amount of time to complete on the server. If you need to take manual control of listening for status updates of the operation, `createExtractJob` should be a better fit.

**Comment by Dsiddhesh** [comment]
Thanks @stephenplusplus  i think  'createExtractJob ' will suffice for us. Could you please share documentation link for it along with example if possible. I am  unable to get it.

**Comment by Dsiddhesh** [comment]
Thanks @stephenplusplus  . 'createExtractJob' works for us.  It will be helpful if you update documentation https://cloud.google.com/bigquery/docs/exporting-data to have a note somewhere mentioning about this method.
